File: Advanced Methods of Programming/Toy Language Interpreter/src/Model/Statements/Composite.java
package Model.Statements;

import Model.Containers.MyIStack;
import Model.ProgramState;

public class Composite implements IStatement {
    private IStatement first;
    private IStatement second;

    public Composite(IStatement first, IStatement second){
        this.first = first;
        this.second = second;
    }

    @Override
    public ProgramState exec(ProgramState state) {
        MyIStack<IStatement> stk = state.getExeStack();
        stk.push(second);
        stk.push(first);
        return null;
    }

    @Override
    public String toString() {
        return first.toString() + ";\n" + second.toString();
    }
}

File: Advanced Methods of Programming/CSharp Toy Language Interpreter/ToyLanguageInterpreter/Src/Model/Statements/Composite.cs
using System;
using ToyLanguageInterpreter.Src.Model;
using ToyLanguageInterpreter.Src.Model.Containers;
using ToyLanguageInterpreter.Src.Model.Statements;

namespace ToyLanguageInterpreter.Src.View
{
    internal class Composite : IStatement
    {
        private IStatement first;
        private IStatement second; 
       
        public Composite(IStatement first, IStatement second)
        {
            this.first = first;
            this.second = second; 
        }

        public ProgramState Exec(ProgramState state)
        {
            IStack<IStatement> stack = state.ExeStack;
            stack.Push(second);
            stack.Push(first);
            return null;
        }

        public override string ToString()
        {
            return first.ToString() + ";\n" + second.ToString();
        }
    }
}